### PR TITLE
fix: error with order by test

### PR DESCRIPTION
**Pull request by massfords**
chore: bump dependencies
chore: update some js-doc
chore: update tsconfig settings
chore: lots of prettier changes due to prettier v3 default change

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,28 +29,28 @@
...
-    "eslint-plugin-prettier": "^4.1.0",
-    "jest": "^28.1.1",
-    "pg-promise": "^10.11.1",
-    "prettier": "^2.7.1",
+    "@types/common-tags": "^1.8.4",
+    "@types/jest": "^29.5.14",
+    "@types/uuid": "^10.0.0",
+    "@typescript-eslint/eslint-plugin": "^8.11.0",
+    "@typescript-eslint/parser": "^8.11.0",
+    "eslint": "^8.57.1",
```
Comment (massfords) [inline package.json:37]: assorted dependency bumps but stayed away from eslint 9 until I have time to adopt the flat config

File: package.json [hunk 2]
```diff
@@ -29,28 +29,28 @@
...
+    "eslint-config-prettier": "^8.10.0",
+    "eslint-plugin-jest": "^28.8.3",
+    "eslint-plugin-prettier": "^5.2.1",
+    "jest": "^29.7.0",
+    "pg-promise": "^11.10.1",
+    "prettier": "^3.3.3",
     "testcontainers": "^9.1.3",
```
Comment (massfords) [inline package.json:44]: deliberately avoided bump to testcontainers. Need to look at what changed with respect to the Postgres support. I think their exports changed but don't have time now to look at it

File: src/llb/to-sql.ts [hunk 3]
```diff
@@ -14,13 +14,13 @@ import { maybeExecuteRsqlOperatorPlugin } from "../plugin";
 /**
  * Formats a keyword according to configuration (either fully upper- or lower-case).
  *
- * @param keyword- The keyword to format.
- * @param [context] - The configuration.
+ * @param keyword - The keyword to format.
+ * @param keywordsLowerCase - The configuration.
```
Comment (massfords) [inline src/llb/to-sql.ts:18]: fix js-doc issue

File: src/plugin.ts [hunk 4]
```diff
@@ -92,7 +92,7 @@ export const isBooleanValueInvariant = (ast: ComparisonNode): void => {
  * > was to have a solution for following `pg` problem: [parameterized query with an
  * > `IN` operator](https://github.com/brianc/node-postgres/issues/1452).
  */
-export const MapInToEqualsAnyPlugin: RsqlOperatorPlugin = {
+export const MapInToEqualsAnyPlugin = {
```
Comment (massfords) [inline src/plugin.ts:95]: best to have this use `satisfies` to avoid having the `invariant` func reported as optional

File: src/plugin.ts [hunk 5]
```diff
@@ -117,7 +117,7 @@ export const MapInToEqualsAnyPlugin: RsqlOperatorPlugin = {
  * > was to have a solution for following `pg` problem: [parameterized query with an
  * > `IN` operator](https://github.com/brianc/node-postgres/issues/1452).
  */
-export const MapOutToNotEqualsAllPlugin: RsqlOperatorPlugin = {
+export const MapOutToNotEqualsAllPlugin = {
```
Comment (massfords) [inline src/plugin.ts:120]: same change with `satisfies` to make the test a little cleaner

File: src/query.ts [hunk 6]
```diff
@@ -32,7 +32,10 @@ export const assembleFullQuery = (
   if (sqlPredicateAndOrderBy.sql === "") {
     return { isValid: true, sql: mainQuery };
   }
-  if (sqlPredicateAndOrderBy.sql.startsWith("ORDER BY" || "order by")) {
+  if (
+    sqlPredicateAndOrderBy.sql.startsWith("ORDER BY") ||
+    sqlPredicateAndOrderBy.sql.startsWith("order by")
+  ) {
```
Comment (massfords) [inline src/query.ts:38]: eslint caught this bug. I missed it in the PR

File: src/tests/plugin.test.ts [hunk 7]
```diff
@@ -178,27 +178,23 @@ describe("tests for sql generation by plugins", () => {
...
     });
 
     describe("invariant", () => {
       it("should fail on falsy", () => {
         expect(() =>
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/ban-types
-          (MapInToEqualsAnyPlugin.invariant as Function)({
-            operands: null,
-          })
+          MapInToEqualsAnyPlugin.invariant({} as unknown as ComparisonNode),
```
Comment (massfords) [inline src/tests/plugin.test.ts:189]: remove the suppressions by changing the declaration of the plugin and using a cast.

File: src/tests/plugin.test.ts [hunk 8]
```diff
@@ -178,27 +178,23 @@ describe("tests for sql generation by plugins", () => {
...
       });
 
       it("should pass on truthy", () => {
         expect(() =>
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/ban-types
-          (MapInToEqualsAnyPlugin.invariant as Function)({
+          MapInToEqualsAnyPlugin.invariant({
             operands: [],
-          })
+          } as unknown as ComparisonNode),
```
Comment (massfords) [inline src/tests/plugin.test.ts:197]: same change as above with respect to the suppressions

File: src/tests/plugin.test.ts [hunk 9]
```diff
@@ -226,27 +222,23 @@ describe("tests for sql generation by plugins", () => {
...
     });
 
     describe("invariant", () => {
       it("should fail on falsy", () => {
         expect(() =>
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/ban-types
-          (MapOutToNotEqualsAllPlugin.invariant as Function)({
-            operands: null,
-          })
+          MapOutToNotEqualsAllPlugin.invariant({} as unknown as ComparisonNode),
```
Comment (massfords) [inline src/tests/plugin.test.ts:233]: and again w/ the suppressions

File: src/tests/plugin.test.ts [hunk 10]
```diff
@@ -226,27 +222,23 @@ describe("tests for sql generation by plugins", () => {
...
       });
 
       it("should pass on truthy", () => {
         expect(() =>
-          // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/ban-types
-          (MapOutToNotEqualsAllPlugin.invariant as Function)({
+          MapOutToNotEqualsAllPlugin.invariant({
             operands: [],
-          })
+          } as unknown as ComparisonNode),
```
Comment (massfords) [inline src/tests/plugin.test.ts:241]: and again w/ the suppressions

File: tsconfig.json [hunk 11]
```diff
@@ -12,7 +12,7 @@
     "disableSourceOfProjectReferenceRedirect": false  /* Disable preferring source files instead of declaration files when referencing composite projects */,
 
     /* Language and Environment */
-    "target": "es6"                                /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
+    "target": "ES2020"                                /* Set the JavaScript language version for emitted JavaScript and include compatible library declarations. */,
```
Comment (massfords) [inline tsconfig.json:15]: needed to bump the target to support the named groups in regex

this was likely due to a bump in the TS version and the compiler being more strict

File: tsconfig.json [hunk 12]
```diff
@@ -30,7 +30,6 @@
     "sourceMap": true                                 /* Create source map files for emitted JavaScript files. */,
     "removeComments": true                            /* Disable emitting comments. */,
     "noEmit": false,                                   /* Disable emitting files from a compilation. */
-    "importsNotUsedAsValues": "error",                /* Specify emit/checking behavior for imports that are only used for types. */
```
Comment (massfords) [inline tsconfig.json:33]: removed after getting yelled at by the compiler

File: package.json (hunk 1 above)
Comment (deadratfink) [reply package.json:37]: Eslint 9 provides a migration target to adapt to new version. I tried that as well, but noticed that it had a bigger  impact with some rule changes and code formattings and would push it to a later story. If you like, I also can do that maybe in the next days and prepare a PR?

**Comment by deadratfink** [comment]
Hi, looks all good to me, can be merged.

File: package.json (hunk 2 above)
Comment (deadratfink) [reply package.json:44]: Yeah, guess that should not be a big issue.

**Review by deadratfink — APPROVED** [review]
Hi, looks all fine, can be merged.

Eslint 9 provides a migration target to adapt to new version. I tried that as well, but noticed that it had a bigger impact with some rule changes and code formattings and would push it to a later story. If you like, I also can do that maybe in the next days and prepare a PR?

**Pull request merged**